### ISSUE TITLE
Audit code_C++

### ISSUE THREAD

**Issue by LifWirser** [vegastrike/Vega-Strike-Engine-Source#16]
This issue is fir finding and correcting C++ issues. Also dicussions on how to test and  measure performance of C++ code.

some issues to be discussed here are
- finding unused/ obsolete/depreciated  C++/h files
- correcting compile errors/warnings
-incorperating  improvments

**Comment by LifWirser** [comment]
I have started to audit C++ #include statements. Basically I have commented out the "#include" statements of one file at a time then trying to compiling. As this produces errors I try to uncomment the needed files neede to solve the first error only. by repeating this process until the compile error are solved, any "include" statements left commented out are unneeded and are removed. then the next file is processed in the same way. Currently I have gotten through >20% of the c++ files Cmake list, After this I'll do the sam for the*.h files . When this is done and tested, I'll create a new repository VS_pre-0.5.4
